fe/fe-core/src/main/java/com/starrocks/transaction/TransactionState.java

```diff
@@ -901,18 +902,16 @@ public boolean allPublishTasksFinishedOrQuorumWaitTimeout(Set<Long> publishError
 
     // Note: caller should hold db lock
     public void prepareFinishChecker(Database db) {
-        if (finishChecker == null) {
-            synchronized (this) {
-                if (finishChecker == null) {
-                    finishChecker = TransactionChecker.create(this, db);
-                }
-            }
+        synchronized (this) {
+            finishChecker = TransactionChecker.create(this, db);
+            checkerCreationTime = System.nanoTime();
         }
     }
 
     public boolean checkCanFinish() {
-        // this may happen if FE restarts
-        if (finishChecker == null) {
+        // finishChecker may be null if FE restarts
+        // finishChecker may require refresh if table/partition is dropped, or index is changed caused by Alter job
```

Comment:
During the checker, if the table is dropping, what will happen?

Reply:
this round of check will check all the tablets before dropping(and probably won't finish), and after next refresh, it will finish